FRFuel.cs

```diff
@@ -88,350 +58,156 @@ public FRFuel()
-                        Pickup pickup = new Pickup(pickupHandle);
-
-                        pickups.Add(pickup);
-                    }
-                });
-            }
-            await Task.FromResult(0);
-        }
-        #endregion
-
-        /// <summary>
-        /// External API for getting current fuel leve;
-        /// </summary>
-        /// <returns></returns>
-        public float ExportsGetCurrentFuelLevel()
-        {
-            if (!PlayerVehicleViableForFuel())
-            {
-                return -1f;
-            }
-
-            return Game.PlayerPed.CurrentVehicle.FuelLevel;
-        }
-
-        /// <summary>
-        /// External API for adding fuel
-        /// </summary>
-        /// <param name="amount"></param>
-        public void ExportsAddFuel(float amount)
-        {
-            if (PlayerVehicleViableForFuel())
-            {
-                var vehicle = Game.PlayerPed.CurrentVehicle;
-
-                VehicleSetFuelLevel(vehicle, vehicle.FuelLevel + amount);
-            }
-        }
-
-        /// <summary>
-        /// External API for setting fuel level
-        /// </summary>
-        /// <param name="amount"></param>
-        public void ExportsSetFuel(float amount)
-        {
-            if (PlayerVehicleViableForFuel())
-            {
-                var vehicle = Game.PlayerPed.CurrentVehicle;
+                    var pos = pickup.Position;
+                    pickup.Delete();
+                    _jerryCans.Add(new Pickup(API.CreatePickup(0xc69de3ff, pos.X, pos.Y, pos.Z - 0.5f, 8 | 32, 0, true, 883325847)));
+                }
 
-                VehicleSetFuelLevel(vehicle, amount);
+                await BaseScript.Delay(1000 * 60 * 10); //Every 10 minutes
             }
         }
 
         /// <summary>
-        /// Returns gas station position that is in range
+        /// On tick
         /// </summary>
-        /// <param name="pos"></param>
-        /// <param name="rangeSquared"></param>
         /// <returns></returns>
-        public int GetGasStationIndexInRange(Vector3 pos, float rangeSquared)
+        public async Task OnTick()
         {
-            for (int i = 0; i < blips.Length; i++)
+            ShowFuelBar();
+            RefuelCheck();
+            if (!Config.GetInstance().AllowManualRefills)
             {
-                Blip blip = blips[i];
-
-                if (Vector3.DistanceSquared(GasStations.positions[i], pos) < rangeSquared)
-                {
-                    return i;
-                }
+                ManualRefuelCheck();
             }
 
-            return -1;
+
+            await Task.FromResult(0);
         }
 
         /// <summary>
-        /// Returns "adequate" vehicle's petrol tank position
+        /// Processes fuel consumption
         /// </summary>
         /// <param name="vehicle"></param>
-        /// <returns></returns>
-        public Vector3 GetVehicleTankPos(Vehicle vehicle)
+        public async Task PeriodicFuelCheck()
```

Comment:
Thanks for pointing out fuel consumption rate dependency on fps, we will fix it.